messages/teleporter/message_manager.go

```diff
@@ -105,6 +108,19 @@ func (m *messageManager) ShouldSendMessage(warpMessageInfo *vmtypes.WarpMessageI
 	if !ok {
 		return false, fmt.Errorf("relayer not configured to deliver to destination. destinationChainID=%s", destinationChainID.String())
 	}
+
+	// If allowedDestinations is empty, then all destinations are allowed
+	// If allowedDestinations is not empty, then only the allowed destinations are allowed
+	if _, exist := m.allowedDestinations[destinationChainID]; !exist && len(m.allowedDestinations) > 0 {
```

Comment:
Also, I think the condition may be wrong. We want to return early here if either 1) `allowedDestinations` is empty OR 2) `allowedDestinations` is non-empty, and the `destinationChainID` is present. I think we want something like:

```
if dsts, exist := m.allowedDestinations[sourceChainID]; exists {
    if allowed, ok := dsts[destinationChainID]; !ok || !allowed {
        m.logger.Info(...)
        return false, nil
    }
}
```

Reply:
Same as above. The `m.allowedDestinations` is map[id.ID]bool and the key is `destinationChainID`.

Reply:
Gotcha. I think we still want to check against the inner condition in my comment above. We need to check the length of the map first though.
```
if len(m.allowedDestinations) > 0 {
    if allowed, ok := m.allowedDestinations[destinationChainID]; !ok || !allowed {
        m.logger.Info(...)
        return false, nil
    }
}
```